src/api/eventTypes.js

```diff
@@ -173,3 +175,253 @@ export type RestartEvent = $ReadOnly<{|
   zulip_version?: string,
   zulip_feature_level?: number,
 |}>;
+
+/* eslint-disable flowtype/generic-spacing */
+
+/**
+ * Data updates we might get in `update` or `update_dict` realm events.
+ *
+ * We expect this to be live-updates to the data we request by putting
+ * 'realm' in `fetch_event_types` when we call POST /register. We type the
+ * shape of that data as InitialDataRealm. The properties are named
+ * differently between here and there (InitialDataRealm has lots of
+ * properties that start with "realm_"), but we expect the values to be the
+ * same types.
+ */
+// If there's a property that exists in InitialDataRealm, but we don't use
+// it and the event doc doesn't mention it, write the InitialDataRealm
+// property in a code comment in the place where it would go here.
+/* prettier-ignore */
+type RealmDataForUpdate = $ReadOnly<{
```

Comment:
Because this type is pretty long and boilerplate-y, and of a different flavor from the event types, I'd like to get it out of this file -- I think that will help keep the rest of the event types more readable.

Perhaps a file `src/api/realmDataTypes.js`?

Then the actual event types would still be in this file -- they'd just import it. This is a bit like how `StreamEvent` refers to the `Stream` type over in `modelTypes.js`.